src/combine.cpp

```diff
@@ -2592,14 +2596,26 @@ void combine_labels(void)
             else if (cur->type == CT_WORD)
             {
                tmp = chunk_get_next_nc(next, CNAV_PREPROC);
+#ifdef DEBUG
+               LOG_FMT(LGUY, "(%d) ", __LINE__);
+#endif
+               LOG_FMT(LGUY, "%s: %d:%d, tmp=%s\n", __func__,
+                       tmp->orig_line, tmp->orig_col, (tmp->type == CT_NEWLINE) ? "<NL>" : tmp->text());
+               log_pcf_flags(LGUY, tmp->flags);
                if (next->flags & PCF_IN_FCN_CALL)
                {
                   /* Must be a macro thingy, assume some sort of label */
                   set_chunk_type(next, CT_LABEL_COLON);
                }
-               else if ((tmp == NULL) || (tmp->type != CT_NUMBER && tmp->type != CT_SIZEOF))
+               else if ((tmp == NULL) ||
+                        (tmp->type != CT_NUMBER &&
+                         tmp->type != CT_SIZEOF &&
```

Comment:
If you are changing these lines could you add some extra parens to be sure we're not shooting ourselves in the foot (I don't _want_ to remember nor to rely on C++ operator precedence - magic stuff).